repository.go

```diff
@@ -573,6 +573,15 @@ func (r *Repository) DeleteBranch(name string) error {
 	return r.Storer.SetConfig(cfg)
 }
 
+// CreateReference creates a reference.
+func (r *Repository) CreateReference(name plumbing.ReferenceName, hash plumbing.Hash) (*plumbing.Reference, error) {
+	ref := plumbing.NewHashReference(name, hash)
+	if err := r.Storer.SetReference(ref); err != nil {
```

Comment:
`ReferenceStorer` also has [CheckAndSetReference(newRef, oldRef)](https://github.com/goabstract/go-git/blob/master/plumbing/storer/reference.go#L19) which checks to make sure an existing ref is not getting overwritten. Would we want to be careful about that? 

It would require another line to get the oldRef, but then you can just pass it straight in and it will do the nil check and whatnot for you.

Reply:
I actually had something similar originally (checking that the ref doesn't already exist), but decided it wasn't really needed. I can put it back though.